spec/decorator.rb

```diff
@@ -0,0 +1,25 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator
+  def correct_name
+    name = @nameable.correct_name.capitalize[0..9]
+    name.length <= @nameable.correct_name.capitalize.length ? "#{name}..." : name
+  end
+end
```

Comment:
Well done so far 👍. Just a few things I would like to recommend 😉.  Just like in my previous comment, I suggest that you make this class extend the Decorator class so that it has access to an instance of the Person class. Your class might look something like this: 

```sh
class TrimmerDecorator < Decorator
  def correct_name
    name = @nameable.correct_name.capitalize[0..9]
    name.length <= @nameable.correct_name.capitalize.length ? "#{name}..." : name
  end
end
```

However, consider this implementation of the ```correct_name``` method in the ```TrimmerDecorator``` class:

```sh
  def correct_name
     @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
  end
```